src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -136,7 +136,7 @@ export class SavedObjectsRepository {
     injectedConstructor: any = SavedObjectsRepository
   ): ISavedObjectsRepository {
     const mappings = migrator.getActiveMappings();
-    const allTypes = Object.keys(getRootPropertiesObjects(mappings));
+    const allTypes = typeRegistry.getAllTypes().map((t) => t.name);
```

Comment:
apparently `Object.entries()` which gets called from inside `getRootPropertiesObjects` is very slow. 

Pierre already refactored many places where we use this round about `getRootPropertiesObjects ` way to get all the types, but there's still a few remaining.

Reply:
Some `getRootPropertiesObjects` were kept because of the `config` 'virtual' type that wasn't properly registered to the registry. But this has been resolved since, so we're safe to remove all the remaining usages.

Very surprised `Object.keys` is a bottleneck, even if it makes sense that `typeRegistry.getAllTypes()` is faster as the registry is based on a `Map`.

Reply:
Probably it's not `Object.keys` but `getRootPropertiesObjects`  mutating an object? v8 cannot optimizie this case https://richardartoul.github.io/jekyll/update/2015/04/26/hidden-classes.html

Reply:
Right, I misread and thought it was just a call to `getRootProperties`

Reply:
This was the line that showed up in the profiler and took almost 5ms: https://github.com/elastic/kibana/blob/ea7c78d10eaf2f80265eea0a564eda0607c58147/src/core/server/saved_objects/mappings/lib/get_root_properties_objects.ts#L46